tools/circle_plus_gen/lib/circle_plus.py

```diff
@@ -67,6 +74,42 @@ def set_train_param(self, train_param: TrainParam):
         tparam_buff = train_param.to_buff()
         self._add_metadata(self.TINFO_META_TAG, tparam_buff)
 
+    def get_number_of_operators(self, subgraph_idx=0) -> int:
+        '''Return a number of operators in the subgraph'''
+        subgraphs: typing.List[cir_gen.SubGraphT] = self.model.subgraphs
+        _not_none(subgraphs, "subgraphs")
+
+        operators: typing.List[cir_gen.OperatorT] = subgraphs[subgraph_idx].operators
+        if operators == None:
+            return 0
+        return len(operators)
+
+    def get_operators(self, subgraph_idx=0) -> typing.List[str]:
+        '''Return a list of opcodes according to the order of operations in the subgraph'''
+        subgraphs: typing.List[cir_gen.SubGraphT] = self.model.subgraphs
```

Comment:
FYI, it can be simple if you just use type hint without `typing` prefix. It would prevent the codes from being too verbose.

```python
from typing import List

subgraphs: List[cir_gen.SubGraphT] = self.model.subgraphs
```

Reply:
:+1: 

To update this, I have to modify another part of the code. 
So, I'll handle this as next PR.